deployment/stacks/ui.ts

```diff
@@ -0,0 +1,244 @@
+        viewerProtocolPolicy: cloudfront.ViewerProtocolPolicy.REDIRECT_TO_HTTPS,
+        cachePolicy,
+        compress: true,
+        responseHeadersPolicy: securityHeadersPolicy,
+      },
+      defaultRootObject: "index.html",
+      enableLogging: true,
+      logBucket,
+      logFilePrefix: `AWSLogs/CLOUDFRONT/${stage}/`,
+      httpVersion: cloudfront.HttpVersion.HTTP2,
+      errorResponses: [
+        {
+          httpStatus: 403,
+          responseHttpStatus: 200,
+          responsePagePath: "/index.html",
+        },
+      ],
+    }
+  );
+
+  distribution.applyRemovalPolicy(
+    isDev ? RemovalPolicy.DESTROY : RemovalPolicy.RETAIN
+  );
+
+  if (!isLocalStack) {
+    const waf = setupWaf(scope, stage, project); // vpnIpSetArn, vpnIpv6SetArn
+    distribution.attachWebAclId(waf.webAcl.attrArn);
+  }
+
+  const applicationEndpointUrl = `https://${distribution.distributionDomainName}/`;
+
+  addIamPropertiesToBucketRole(
+    scope,
+    "Custom::S3AutoDeleteObjectsCustomResourceProvider/Role",
+    iamPermissionsBoundary.managedPolicyArn,
+    iamPath
+  );
+
+  return {
+    cloudfrontDistributionId: distribution.distributionId,
+    distribution,
+    applicationEndpointUrl,
+    s3BucketName: uiBucket.bucketName,
+    uiBucket,
+  };
+}
+
+function setupWaf(
+  scope: Construct,
+  stage: string,
+  project: string
+  // vpnIpSetArn?: string,
+  // vpnIpv6SetArn?: string,
+) {
+  return new WafConstruct(
+    scope,
+    "CloudfrontWafConstruct",
+    {
+      name: `${project}-${stage}-ui`,
+      blockByDefault: false,
+    },
+    "CLOUDFRONT"
+  );
+  // Additional Rules for this WAF only if CMS asks to have the application made vpn only
+  // const wafRules: wafv2.CfnWebACL.RuleProperty[] = [];
+
+  // const defaultAction = vpnIpSetArn
+  //   ? { block: {} }
+  //   : { allow: {} };
+
+  // if (vpnIpSetArn) {
+  //   const githubIpSet = new wafv2.CfnIPSet(scope, "GitHubIPSet", {
+  //     name: `${stage}-gh-ipset`,
+  //     scope: "CLOUDFRONT",
+  //     addresses: [],
+  //     ipAddressVersion: "IPV4",
+  //   });
+
+  //   const statements = [
+  //     {
+  //       ipSetReferenceStatement: { arn: vpnIpSetArn },
+  //     },
+  //     {
+  //       ipSetReferenceStatement: { arn: githubIpSet.attrArn },
+  //     },
+  //   ];
+
+  //   if (vpnIpv6SetArn) {
+  //     statements.push({
+  //       ipSetReferenceStatement: {
+  //         arn: vpnIpv6SetArn,
+  //       },
+  //     });
+  //   }
+
+  //   wafRules.push({
+  //     name: "vpn-only",
+  //     priority: 0,
+  //     action: { allow: {} },
+  //     visibilityConfig: {
+  //       cloudWatchMetricsEnabled: true,
+  //       metricName: `${project}-${stage}-webacl-vpn-only`,
+  //       sampledRequestsEnabled: true,
+  //     },
+  //     statement: {
+  //       orStatement: {
+  //         statements,
+  //       },
+  //     },
+  //   });
+  // }
+  distribution.applyRemovalPolicy(RemovalPolicy.RETAIN);
```

Comment:
I think this is set in a different part of this file based on isDev which is probably what it should be.

Reply:
Fixed, thanks!